modules/340-monitoring-kubernetes/hooks/helm.go

```diff
@@ -369,6 +413,9 @@ func (uvs unsupportedVersionsStore) CalculateCompatibility(currentVersion *semve
 		}
 	}
 
+	// check if the api was the closest k8s version restriction
+	unsupportVersion.verify(resourceAPIVersion, resourceKind)
```

Comment:
You have to also check `Automatic` kubernetes mode here. Otherwise it will block releases even with pinned k8s version

Reply:
Check kubernetes mode added.